### PR TITLE
fix: add a generic cleaning mechanism

### PR DESCRIPTION
**Pull request by morgante**
Fixes https://github.com/getgrit/gritql/issues/209 by replacing our special-cased "orphan" logic with a generic cleaning approach where we can repair program output.

## Alternatives considered
I tried doing this as part of after_each_file but that happens pre-linearization and making additional changes there was too cumbersome: https://github.com/getgrit/stdlib/pull/187

## Summary by CodeRabbit

- **New Features**
  - Introduced a new system for managing code replacements across various programming languages, enhancing the ability to handle and apply changes within source code efficiently.

- **Refactor**
  - Transitioned from handling "orphaned" code elements to a more robust "replacement" system, affecting multiple language implementations including JavaScript, Python, TypeScript, and TSX.
  - Updated internal logic and method signatures to support the new replacement system.

- **Tests**
  - Added tests to ensure the new replacement functionalities work correctly for JavaScript code.

- **Bug Fixes**
  - Improved handling of specific node replacements in scripts, ensuring more accurate code modifications.

### CODE REVIEW

File: crates/core/src/test.rs [hunk 1]
```diff
@@ -7960,6 +7960,37 @@
...
+                |const fn = () =>{} ;
+                |const fnTwo = () => {  };
+                |const fnBob = () => { alert(); }
+                |"#
+            .trim_margin()
+            .unwrap(),
+        }
+    })
+    .unwrap();
+}
```
[the remark this answers is not in the document]
Comment (morgante) [reply crates/core/src/test.rs:7992]: Wrong, this is intentional.

**Review by seren5240 — APPROVED** [review]

**Comment by scr2em** [comment]
@morgante Does this cover a case like this? 
```js
const x = () => (log(), log());
``` 
query
```
`log()` => .
```

should be 
```js
const x = () => {}
```

I know it's a very weird case, but just got it in mind. Ignore if it's too trivial

**Comment by morgante** [comment]
@scr2em That input isn't valid JavaScript (so Grit isn't expected to handle it properly), but this works fine: `const x = () => {console.log(); console.log()};`

**Comment by morgante** [comment]
See https://github.com/getgrit/gritql/pull/212/files

**Comment by scr2em** [comment]
@morgante I believe it's a valid javascript, can you recheck ? 
```js 
const x = () => (log(), log());
```

image

**Comment by scr2em** [comment]
just noticed that we can just remove the assign operator

**Pull request merged**